presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1476,6 +1478,18 @@ else if (element instanceof Rollup) {
 
             if (node.getGroupBy().isPresent()) {
                 checkGroupingSetsCount(node.getGroupBy().get());
+
+                // Only SimpleGroupBy allows complex expressions (a deviation from the SQL specification)
```

Comment:
move this directly above `.filter(element -> !(element instanceof SimpleGroupBy))`